packages/cli/templates/react/src/components/FieldPluginExample/index.spec.tsx

```diff
@@ -1,43 +1,166 @@
-import { FieldPluginResponse } from '@storyblok/field-plugin'
-
-vi.mock('@storyblok/field-plugin/react')
-
-const fieldPluginDefault: FieldPluginResponse<number | undefined> = {
-  type: 'loaded',
-  data: {
-    isModalOpen: false,
-    content: undefined,
-    options: {},
-    spaceId: undefined,
-    storyLang: '',
+
+const fakeContainer = (listener) => {
+  const schema = {
+    field_type: 'test-field-plugin',
+    options: [],
+  }
+  let height = undefined
+  let uid = 'test-uid'
+  let model = undefined
+  let isModalOpen = false
+  let blockId = 'test-block-id'
+  let language = 'default'
+  let storyId = 'test-story-id'
+  let spaceId = 'test-space-id'
+  let token = 'test-token'
+
+  const stateMessage = ({ action, callbackId }) => ({
+    callbackId: callbackId,
+    schema,
+    model,
+    isModalOpen,
+    uid,
+    blockId,
     story: {
       content: {},
     },
-    storyId: undefined,
-    blockUid: undefined,
-    token: undefined,
-    uid: '',
-  },
-  actions: {
-    setContent: vi.fn(),
-    setModalOpen: vi.fn(),
-    requestContext: vi.fn(),
-    selectAsset: vi.fn(),
-  },
+    language,
+    storyId,
+    spaceId,
+    token,
+    action,
+  })
+
+  return {
+    got: ({ data, origin }) => {
+      if (isPluginLoadedMessage(data)) {
+        listener(
+          stateMessage({
+            action: 'loaded',
+            callbackId: data.callbackId,
+          }),
+        )
+      } else if (isValueChangeMessage(data)) {
+        model = data.model
+        listener(
+          stateMessage({
+            action: 'state-changed',
+            callbackId: data.callbackId,
+          }),
+        )
+      } else if (isModalChangeMessage(data)) {
+        isModalOpen = data.status
+        listener(
+          stateMessage({
+            action: 'state-changed',
+            callbackId: data.callbackId,
+          }),
+        )
+      } else if (isHeightChangeMessage(data)) {
+        height = data.height
+        listener(
+          stateMessage({
+            action: 'state-changed',
+            callbackId: data.callbackId,
+          }),
+        )
+      } else if (isAssetModalChangeMessage(data)) {
+        listener({
+          action: 'asset-selected',
+          uid,
+          filename: 'https://...',
+          field: '',
+          callbackId: data.callbackId,
+        })
+      } else if (isGetContextMessage(data)) {
+        listener({
+          action: 'get-context',
+          uid,
+          callbackId: data.callbackId,
+          story: {
+            content: {},
+          },
+        })
+      } else {
+        console.warn(
+          `Container received unknown message from plugin: ${JSON.stringify(
+            data,
+          )}`,
+        )
+      }
+    },
+  }
+}
+
+const setup = () => {
+  let messageCallback
+  const listener = (data) => {
+    messageCallback({ data })
+  }
+  const container = fakeContainer(listener)
+  global.ResizeObserver = class ResizeObserver {
```

Comment:
We could extract the observer mocking in a separate method, to shorten the setup function a bit.